esmvalcore/cmor/check.py

```diff
@@ -647,7 +645,8 @@ def _check_coord_points(self, coord_info, coord, var_name):
                                          valid_max)
 
         if l_fix_coord_value:
-            if coord.ndim == 1:
+            # cube.intersection only works for cells with 0 or 2 bounds
+            if coord.ndim == 1 and coord.nbounds in (0, 2):
```

Comment:
What's `nbounds` on longitudes for icon? If it's neither 0 nor 2, is there an easy way to understand it?

Reply:
`nbounds=3` for ICON, which correspond to the vertices of triangles, see

![grafik](https://user-images.githubusercontent.com/32543114/133616935-1a458d2e-1fcc-4322-8b72-abcc307cb7ab.png)

Reply:
Oh yeah, of course. In that case, I suggest adding a brief comment to the effect of:
> `nbounds==0` means there are no bounds given, `nbounds==2` implies a regular grid with bounds in the grid direction, `nbounds>2` implies an irregular grid with bounds given as vertices of the cell polygon.